src/proto/robot_config.proto

```diff
@@ -18,7 +18,110 @@ package bowler.robot_config;
 
 option java_multiple_files = true;
 option java_package = "com.commonwealthrobotics.proto.robot_config";
+//next 2
+// a center of mass object
+message CenterOfMass{
+  Pose centroid = 1;
+  double mass_kg = 2;
+}
 
-// Next: 1
+// next 3
+// a file location of a provider of some depency to be injected
+message GitFile {
+  string url = 1;
+  string branch = 2;
+  string file = 3;
+}
+// next 10
+message LimbType{
+  bool isArm = 1;
+  bool isLeg = 2;
+  bool isSteerableWheel = 3;
+  bool isHead = 4;
+  bool isTail = 5;
+  bool isWing = 6;
+  bool isThruster = 7;
+  bool isFin = 8;
+  bool isSpine = 9;
+  bool isFixedWheel=10;
+}
+//next 3
+message Vitamin{
+  string type = 1;
+  string size = 2;
+  CenterOfMass com = 3;
+}
+
+// next 7
+// A pose in 6dof cartesian space
+message Pose {
+  double x_mm = 1;
+  double y_mm = 2;
+  double z_mm = 3;
+  double qw = 4;
+  double qx = 5;
+  double qy = 6;
+  double qz = 7;
+}
+// next 7
+message ParallelGroupData{
+  string parallelGroupName = 4;
+  Pose parallelGroupTipOffset = 5;
+  string relativeToLimbName = 6;
+  uint32 relativeToIndex = 7;
+}
+
+// Next: 8
 // Robot configuration file object
-message RobotConfig {}
+message RobotConfig {
+  repeated Limb appendage = 1;
+  Pose robotZeroToFiducialTransform = 2;
+  string name = 3;
+  repeated GitFile cad = 4;
```

Comment:
What is this for?

Reply:
Ill swap them for the existing definition, but i was using GitFIle sections to represent the locations of dependency for injection on load time. i intend to make the interface take a copy of the config, the limb name and the link index as inputs, and return a list of CSG back, so a simple cad script is valid here. I intend to allow N scripts to be loaded to make adding parts easier than Legacy version.